internal/logging/file_test.go

```diff
@@ -0,0 +1,31 @@
+package logging
+
+import (
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEmit_Parallel(t *testing.T) {
+	var messages []string
+	for i := 0; i < 10; i++ {
+		messages = append(messages, fmt.Sprintf("%d", i))
+	}
+
+	fh := newFileHandler()
+	defer fh.Close()
+
+	loggingFile, err := ioutil.TempFile("", "")
+	assert.NoError(t, err)
+	fh.file = loggingFile
+
+	for i, message := range messages {
+		t.Run(fmt.Sprintf("Parallel emit %d", i), func(t *testing.T) {
+			t.Parallel()
+			emitErr := fh.Emit(&MessageContext{}, message)
+			assert.NoError(t, emitErr)
+		})
```

Comment:
nit: It might be worth putting this in another loop (maybe 10 count) to ensure that the window of time that the detector is sensitive to is landed on.